### PR TITLE
SPARK-5019 [MLlib] - GaussianMixtureModel exposes instances of MultivariateGauss...

### PR DESCRIPTION
**Pull request by tgaloppo**
This PR modifies GaussianMixtureModel to expose instances of MutlivariateGaussian rather than separate mean and covariance arrays.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixtureModel.scala [hunk 1]
```diff
@@ -37,8 +37,9 @@ import org.apache.spark.mllib.util.MLUtils
  */
 class GaussianMixtureModel(
   val weight: Array[Double], 
-  val mu: Array[Vector], 
-  val sigma: Array[Matrix]) extends Serializable {
+  val gaussian: Array[MultivariateGaussian]) extends Serializable {
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixtureModel.scala]: Should "weight" and "gaussian" be plural ("weights" and "gaussians")?

**Comment by jkbradley** [comment]
@tgaloppo Other than the 1 comment, this looks good.  Thanks!

**Comment by tgaloppo** [comment]
@jkbradley I considered making those plural for the initial commit.  I guess I should have.  Update has been made.

**Comment by jkbradley** [comment]
@tgaloppo Thank you!  LGTM

CC: @mengxr

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**